File: app/src/sample/java/com/mycompany/myapp/app/VariantModule.java
package com.mycompany.myapp.app;

import android.support.annotation.Nullable;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLContext;
import javax.net.ssl.X509TrustManager;

import dagger.Module;
import dagger.Provides;

@Module
public class VariantModule {

    @Nullable
    @Provides
    public HostnameVerifier provideCustomHostnameVerifier(Settings settings) {
        if (settings.getTrustAllSSL()) {
            return SSLDevelopmentHelper.buildTrustAllHostnameVerifer();
        } else {
            return null;
        }
    }

    @Nullable
    @Provides
    public SSLContext provideSSLContext(Settings settings) {
        if (settings.getTrustAllSSL()) {
            return SSLDevelopmentHelper.buildTrustAllSSLContext();
        } else {
            return null;
        }
    }

    @Nullable
    @Provides
    public X509TrustManager buildTrustAllTrustManager(Settings settings) {
        if (settings.getTrustAllSSL()) {
            return SSLDevelopmentHelper.buildTrustAllTrustManager();
        } else {
            return null;
        }
    }
}

File: app/src/dev/java/com/mycompany/myapp/app/VariantModule.kt
package com.mycompany.myapp.app

import dagger.Module
import dagger.Provides
import javax.net.ssl.HostnameVerifier
import javax.net.ssl.SSLContext
import javax.net.ssl.X509TrustManager

@Module
class VariantModule {

    @Provides
    fun provideCustomHostnameVerifier(settings: Settings): HostnameVerifier? {
        return when {
            settings.trustAllSSL -> SSLDevelopmentHelper.buildTrustAllHostnameVerifier()
            else -> null
        }
    }

    @Provides
    fun provideSSLContext(settings: Settings): SSLContext? {
        return when {
            settings.trustAllSSL -> SSLDevelopmentHelper.buildTrustAllSSLContext()
            else -> null
        }
    }

    @Provides
    fun buildTrustAllTrustManager(settings: Settings): X509TrustManager? {
        return when {
            settings.trustAllSSL -> SSLDevelopmentHelper.buildTrustAllTrustManager()
            else -> null
        }
    }
}
